src/components/ModalPayment/ModalBuyMeal/ModalBuyMealPool.tsx

```diff
@@ -59,17 +62,22 @@ export const Modal = (props: Props) => {
     // eslint-disable-next-line
   }, []);
 
-  const fetchData = async (sellerId: string) => {
+  const fetchData = async (sellerId: string, campaignId?: string) => {
     // Note(wilsonj806) Showing the campaign that expires first
     // will need to update this if we render multiple distributors
     const { data } = await getCampaignsForMerchant(sellerId);
-    const { data: distrib } = await getDistributor(data[0].distributor_id);
+
+    const campaign = data.find((ele) => campaignId === ele.id);
```

Comment:
Why is `campaignId` optional and wouldn't this blow up if it doesn't get passed in?

Reply:
The `useEffect` that it's called in on line 77 addresses that, but I'll try fix it so it looks less jank.